src/Compilers/Core/Portable/ConstantValue.cs

```diff
@@ -439,6 +439,7 @@ public object Value
                 {
                     case ConstantValueTypeDiscriminator.Bad: return null;
                     case ConstantValueTypeDiscriminator.Null: return null;
+                    case ConstantValueTypeDiscriminator.DefaultLiteral: return null;
```

Comment:
I'm having some more trouble with constant values. The trouble is that expressions like `default(int?)` and `default(int*)` don't have constant values (see [GetDefaultValue](http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp/Symbols/TypeSymbolExtensions.cs,264ab804a03fe275)).

That means that `default` in `int? x = default;` shouldn't have a constant value. Does that seem right?

Also, I explored binding `default` as a BoundConversion with a null literal conversion (instead of creating a new conversion kind) and a BoundLiteralExpression as operand. But I think that literals should have a constant value. Is that correct?